utils/ast_printer/ast_printer.cc

```diff
@@ -0,0 +1,284 @@
+	};
+
+	std::string
+	print(const std::vector<sa::function>& functions) noexcept
+	{
+		std::stringstream ast_stream;
+
+		for (const auto& fun : functions)
+		{
+			textbox graph;
+			graph.putbox(0, 0,
+				create_tree_graph(
+					/* vertex data */ fun.body(),
+					/* line limit */ 100, create_vertex,
+					vertex_children, isnot_oneliner, is_simple,
+					should_separate_first));
+
+			ast_stream << BoldCyan << "\n" << fun.name() + ":\n" << graph.to_string();
+		}
+
+		return ast_stream.str();
+	}
+
+	std::string
+	print(const sa::function& fun) noexcept
+	{
+		return print(fun.body(), /* is_statement = */ true);
+	}
+
+// Declare this for later usage
+	static std::string next_expression(const sa::expression& expr);
+
+	std::string
+	print(const sa::expression& expr, bool is_statement /* = false */) noexcept
+	{
+		// Iterators pointing to the first and last expression in `expr.operands()`
+		const auto& front = expr.operands().cbegin();
+		const auto& back = expr.operands().cend();
+
+		switch (expr.get_type())
+		{
+		case et::Nop:
+			return "";
+		case et::String:
+			return "\"" + std::get<std::string>(expr.terminal()) + "\"";
+		case et::Number:
+			return std::to_string(std::get<int>(expr.terminal()));
+		case et::Identifier:
+			return print(std::get<sa::identifier>(expr.terminal()));
+		case et::Addition:
+			return print(expr, " + ", "()");
+		case et::Multiplication:
+			return print(expr, " * ", "()");
+		case et::Division:
+			return print(expr, " / ", "()");
+		case et::Equality:
+			return print(expr, " == ", "()");
+		case et::ModularDivision:
+			return print(expr, " % ", "()");
+		case et::If:
+			return print(expr, " if ", "()");
+		case et::Negation:
+			return "-(" + next_expression(expr) + ")";
+		case et::Sequence:
+			if (is_statement)
+				return print(expr, "; ", "{}", true);
+			else
+				return print(expr, ", ", "()");
+		case et::Copy:
+		{
+			if (expr.operands().empty())
+				return "??";
+			return "(" + print(*back) + " = " + print(*front) + ")";
+		}
+		case et::While:
+		{
+			const std::string& condition
+				= print(*expr.peculiar().value());
+			const std::string& body = print(expr, ", ", "()", false, 1);
+			return "while " + condition + " " + body;
+		}
+		case et::FunctionCall:
+		{
+			if (expr.operands().empty())
+				return "??";
+			return "( " + print(*front) + " )"
+				+ print(expr, ", ", "()", false, 1);
+		}
+		case et::Return:
+			return "return ( " + next_expression(expr) + " )";
+		}
+		return "????";
+	}
+
+	std::string
+	print(const sa::identifier& ident) noexcept
+	{
+		static std::string prefixes = "??FPV";
+		std::stringstream sstr{};
+		sstr << prefixes[static_cast<int> (ident.get_type())]
+			 << std::to_string(ident.index());
+		sstr << "\"" << ident.name() << "\"";
+
+		return sstr.str();
+	}
+
+	std::string
+	print(const sa::expression& expr, const std::string& oper,
+		const std::string& separator, bool is_statement /* = false */,
```

Comment:
Put this default parameterizing convention into the conventions file

Reply:
I do not think it is necessary, since this is a personal preference for me. I am unsure whether this should be applied as a default practice.